### ISSUE TITLE
[BUG] Unable to enable and configure flow log for network security group

### ISSUE THREAD

**Issue by paulojohnj** [Azure/azure-libraries-for-java#767]
**Describe the bug**
Unable to enable and configure flow log for network security group, using a storage account in either the NetworkWatcherRG or another existing resource group

***Exception or Stack Trace***
The client has permission to perform action 'Microsoft.OperationalInsights/workspaces/sharedKeys/action' on scope '/subscriptions/{subscription_id}/resourceGroups/NetworkWatcherRG/providers/Microsoft.Network/networkWatchers/NetworkWatcher_usgovvirginia', however the linked subscription 'resourcegroups' was not found

**To Reproduce**
- retrieve network watchers
- for network watcher in correct region
- retrieve flow settings for existing network security group in the region
- update flow settings to enable logging and set storage to existing storage account

***Code Snippet***
```
final PagedList<NetworkWatcher> nws = adapter.getItsAzure().networkWatchers().list();
            
            NetworkWatcher retval = null;
            for(final NetworkWatcher nw : nws ) {
                if(nw.region().equals(Region.GOV_US_VIRGINIA)) {
                    retval = nw;
                }
            }
            
            final ResourceGroup rg = adapter.getItsAzure().resourceGroups().getByName(retval.resourceGroupName());
            final StorageAccount sa = adapter.getItsAzure().storageAccounts().define(ResourceNameType.STORAGE_ACCOUNT.randomName("networkwatchersa"))
                .withRegion(Region.GOV_US_VIRGINIA)
                .withExistingResourceGroup(rg)
                .withAccessFromAllNetworks()
                .create();
            
            final String rgName = "resource-group-38f6628eccb84ec9aa1cd9b3c8f5f815";
            final NetworkSecurityGroup nsg = adapter.getItsAzure().networkSecurityGroups().getByResourceGroup(rgName, "add-network1-nat-securitygroup");
            
            final FlowLogSettings fls = retval.getFlowLogSettings(nsg.id());
                
            LOGGER.info("Found fls with enabled {} and storage id {}", fls.enabled(), fls.storageId());
            
            fls.update()
                .withLogging()
                .withStorageAccount(sa.id())
                .apply();
```

**Expected behavior**
Expect to be able to enable flow logs for the nsg in a storage account, or a more elaborate error message, I cannot currently determine what the issue is

**Setup (please complete the following information):**
 - OS: macOS
 - IDE : Eclipse Version: 2019-06 (4.12.0)
 - Version of the Library used: 1.22

**Additional context**
Call has been attempted with the Service Principal as both a contributor and owner in the subscription. I am trying to understand the error message as the sdk call seems straight forward. I suspect it is a permissions or ownership issue.

Update: confirmed this is still and issue in version 1.23

**Comment by kurtzeborn** [comment]
Thank you for opening this issue! We are routing it to the appropriate team for follow up.

**Comment by paulojohnj** [comment]
For what it is worth, it seems I am able to configure a flow log for a network security group in version 1.23 if I specify the line: `fls.inner().withFlowAnalyticsConfiguration(null);` before attempting to update the flow log setting for that particular nsg.

**Comment by weidongxu-microsoft** [comment]
Hello @paulojohnj 

You are exactly right. The underlying bug is that workspaceResourceId (i.e. log analytics workspace) in flowAnalyticsConfiguration is marked as required, even when flowAnalyticsConfiguration is not enabled.

If you do not use flowAnalyticsConfiguration at all, your workaround `fls.inner().withFlowAnalyticsConfiguration(null);` would work fine.

If you use flowAnalyticsConfiguration, after you configured it with log analytics workspace, you will not encounter further error without walkaround.

We are checking internally on the API spec and service implementation (https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcher.json#L2773), to see what is the best way to solve this bug.

**Comment by weidongxu-microsoft** [comment]
To service team:

Network API spec marks property **workspaceResourceId** as required. https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/networkWatcher.json#L2773

However response from the endpoint does not have this property.
```
POST https://management.azure.com/subscriptions/1c638cf4-608f-4ee6-b680-c329e824c3a8/resourceGroups/NetworkWatcherRG/providers/Microsoft.Network/networkWatchers/NetworkWatcher_eastus2/queryFlowLogStatus
{
  "targetResourceId": "/subscriptions/1c638cf4-608f-4ee6-b680-c329e824c3a8/resourceGroups/rg-weidxu/providers/Microsoft.Network/networkSecurityGroups/nsg-weidxu",
  "properties": {
    "storageId": "",
    "enabled": false,
    "retentionPolicy": {
      "days": 0,
      "enabled": false
    }
  },
  "flowAnalyticsConfiguration": {
    "networkWatcherFlowAnalyticsConfiguration": {
      "enabled": false,
      "workspaceId": "-",
      "workspaceRegion": "-",
      "trafficAnalyticsInterval": 60
    }
  }
}
```

When user updates it with enabled=true but flowAnalyticsConfiguration.networkWatcherFlowAnalyticsConfiguration.enabled=false (hence user is not expected to provide workspaceResourceId), it causes SDK exception when updating the resource.

**Comment by yaohaizh** [comment]
As confirmed, this is Swagger issue and need fix from service team.

**Comment by weidongxu-microsoft** [comment]
https://github.com/Azure/azure-rest-api-specs/pull/7080

**Comment by weidongxu-microsoft** [comment]
Fixed in 1.27. To be release on Oct. 11th.
